main_test.go

```diff
@@ -0,0 +1,75 @@
+package meilisearch
+
+import (
+	"os"
+	"testing"
+)
+
+type docTest struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type docTestBooks struct {
+	BookID int    `json:"book_id"`
+	Title  string `json:"title"`
+	Tag    string `json:"tag"`
+}
+
+func deleteAllIndexes(client *Client) (ok bool, err error) {
+	list, err := client.Indexes().List()
+
+	if err != nil {
+		return false, err
+	}
+
+	for _, index := range list {
+		client.Indexes().Delete(index.UID)
+	}
+
+	return true, nil
+}
+
+func TestMain(m *testing.M) {
```

Comment:
Where is this function used?

Reply:
Every time you run `go test` for 'meilisearch' module, it will run `TestMain` which will delete all indexes, run the test as usual and finally delete all indexes again.

So basically, it is triggered by `go test`, `go test -v`, `go test -v ./...`

Reply:
Oh magic!!